include/libkrun.h

```diff
@@ -113,6 +120,41 @@ int32_t krun_set_data_disk(uint32_t ctx_id, const char *disk_path);
  */
 int32_t krun_add_disk(uint32_t ctx_id, const char *block_id, const char *disk_path, bool read_only);
 
+/* Supported disk image formats */
+#define KRUN_DISK_FORMAT_UNSPECIFIED 0
+#define KRUN_DISK_FORMAT_RAW 1
+#define KRUN_DISK_FORMAT_QCOW2 2
+/**
+ * Adds a disk image to be used as a general partition for the microVM. The supported
+ * image formats are: "raw" and "qcow2". 
+ *
+ * This API is mutually exclusive with the deprecated krun_set_root_disk and
+ * krun_set_data_disk methods and must not be used together.
+ *
+ * By all possible means, one should avoid probing the disk image that will be given to libkrun.
+ * If a raw file is probed, and a QCOW2 header is written into the raw file's header, the next time
```

Comment:
For clarity, I would explicitly mention it is the guest that could override the header.
```suggestion
 * If a raw file is probed, and a QCOW2 header is written into the raw file's header by the guest, the next time
```

Reply:
Fixed as well